clicommand/agent_start.go

```diff
@@ -315,8 +321,8 @@ var AgentStartCommand = cli.Command{
 			cfg.BootstrapScript = fmt.Sprintf("%s bootstrap", shellwords.Quote(os.Args[0]))
 		}
 
-		// Turning off command eval will also turn off plugins.
-		if cfg.NoCommandEval {
+		// Turning off command eval or local hooks will also turn off plugins.
+		if cfg.NoCommandEval || cfg.NoLocalHooks {
```

Comment:
Hrm...what's your thinking here?

Reply:
What's the point of disabling local hooks if you can use plugins? You'd just be able to modify pipeline.yml and achieve the same thing as local hooks otherwise.

Reply:
Local hooks by themselves aren't dangerous because you need source code access to change them. Editing a local hook is the same as editing a build script.

The point behind `no-command-eval` is stopping Buildkite executing arbitrary commands on an agent. Since you can define plugins in the BK user interface, you could circumvent `no-command-eval` (which is why we turn off plugins if `no-command-eval` is on).

So I don't think we need to disable plugins if `no-local-hooks` is on? Unless I'm missing another attack vector?

Reply:
My understanding was that the point behind `no-local-hooks` is to prevent hooks in a repository being able to override the hooks set at the global level. They are for where you don't trust the source code, for instance third-party pull requests.

Reply:
Did you have a different understanding of their purpose?

Reply:
Ok, spoke about on video with @lox. If you consider plugins to be "3rd party local hooks" then this makes total sense...!